### PR TITLE
Filehandler

### PR DESCRIPTION
**Pull request by joerivanarkel**

### CODE REVIEW

File: Common/FileReadHandler.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    // Location: Without root folder, i.e. "./"
+    public string ReadAllTextFromFile(string name, string extension, string location)
+    {
+        if (name == string.Empty || extension == string.Empty) { throw new ArgumentNullException("Name, Text or Extension provided is empty"); }
+
+        name += extension;
+        var targetFolder = Path.Combine(Environment.CurrentDirectory, location);
+        string fileName = Path.Combine(targetFolder, name);
+
+        return File.ReadAllText(fileName);
```
Comment (teusvanarkel) [inline Common/FileReadHandler.cs:21]: Check of file exists

File: Common/FileReadHandler.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+namespace Common;
+
+[Serializable]
+public class FileReadHandler : IFileReadHandler
+{
+
+    // Name: Without extension
+    // Extension: With! dot
+    // Location: Without root folder, i.e. "./"
+    public string ReadAllTextFromFile(string name, string extension, string location)
```
Comment (teusvanarkel) [inline Common/FileReadHandler.cs:13]: Change to async version

File: Common/FileWriteHandler.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+namespace Common;
+
+[Serializable]
+public class FileWriteHandler : IFileWriteHandler
+{
+
+    // Name: Without extension
+    // Extension: With! dot
+    // Location: Without root folder, i.e. "./"
+    public bool WriteToFile(string name, string extension, string text, string location, bool encryption = false)
```
Comment (teusvanarkel) [inline Common/FileWriteHandler.cs:12]: Make litle object for all those parameters

File: Common/FileWriteHandler.cs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    private void Write(string name, string extension, string text, string location)
+    {
+        name += extension;
+        var targetFolder = Path.Combine(Environment.CurrentDirectory, location);
+
+        string fileName = Path.Combine(targetFolder, name);
+
+        File.WriteAllText(fileName, text);
```
Comment (teusvanarkel) [inline Common/FileWriteHandler.cs:66]: Change to async

File: Common/FileWriteHandler.cs [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    private void AppendWrite(string name, string extension, string text, string location)
+    {
+        name += extension;
+        var targetFolder = Path.Combine(Environment.CurrentDirectory, location);
+
+        string fileName = Path.Combine(targetFolder, name);
+        
+        var file = File.AppendText(fileName);
```
Comment (teusvanarkel) [inline Common/FileWriteHandler.cs:76]: Change to async

File: Common/Logger.cs [hunk 6]
```diff
@@ -9,29 +9,26 @@ public class Logger : ILogger
     private string logName { get; set; }
     private string version { get; set; } = EnvironmentVariables.GetVariable("version");
     private string location { get; set; }
+    private IFileWriteHandler _fileHandler;
 
-    public Logger(string Name = "placeholder")
+    public Logger(IFileWriteHandler FileHandler, string Name = "")
```
Comment (teusvanarkel) [inline Common/Logger.cs:14]: LowerCase?

File: Common/Logger.cs [hunk 7]
```diff
@@ -9,29 +9,26 @@ public class Logger : ILogger
     private string logName { get; set; }
     private string version { get; set; } = EnvironmentVariables.GetVariable("version");
     private string location { get; set; }
+    private IFileWriteHandler _fileHandler;
```
Comment (teusvanarkel) [inline Common/Logger.cs:12]: make readonly

File: Common/Logger.cs [hunk 8]
```diff
@@ -9,29 +9,26 @@ public class Logger : ILogger
...
 
     public void Log(string message, string orginFile, LogType logType)
     {
-        var file = File.AppendText(location);
         var time = DateTime.Now.ToString();
-        file.Write($"{time.Replace(" uur", "")}: {logType.ToString()}: {orginFile}.cs: {message}\n");
-        file.Close();
+        var text = ($"{time.Replace(" uur", "")}: {logType.ToString()}: {orginFile}.cs: {message}\n");
+
+        _fileHandler.AppendToFile(logName, ".log", text, location);
```
Comment (teusvanarkel) [inline Common/Logger.cs:32]: Make an enum with extensions

**Comment by joerivanarkel** [comment]
**Summary**:
- Added a two filehandlers for writing and reading

**Pull request merged**